NuWeather/plugin.py

```diff
@@ -352,18 +380,22 @@ def _apixu_fetcher(self, location):
                           'summary': forecastdata['day']['condition']['text']} for idx, forecastdata in enumerate(data['forecast']['forecastday'])]
         }
 
-    def _darksky_fetcher(self, location):
+    def _darksky_fetcher(self, channel, location):
         """Grabs weather data from Dark Sky."""
         apikey = self.registryValue('apikeys.darksky')
         if not apikey:
             raise callbacks.Error(_("Please configure the Dark Sky API key in plugins.nuweather.apikeys.darksky."))
 
         # Convert location to lat,lon first
-        latlon = self._geocode(location)
+        latlon = self._geocode(channel, location)
         if not latlon:
             raise callbacks.Error("Unknown location %s." % location)
 
         lat, lon, display_name, geocodeid = latlon
+        if 'osm_id' in geocodeid:
+            geocode_api = 'OSM/Nominatim'
+        else:
+            geocode_api = 'GoogleMaps'
```

Comment:
What I meant was `result = (lat, lon, display_name, place_id, geocode_backend_name)` in each geocode function, so that there's no guessing involved.

Reply:
Alright.

Reply:
Ok, see what you think of the latest change